Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Call.hs

```diff
@@ -104,7 +113,41 @@ initiateCallToCustomer rideId = do
         SCS.CallStatus
           { id = callStatusId,
             callId = exotelResponse.callId,
-            rideId = rideId,
+            rideId = Just rideId,
+            rcId = Nothing,
+            dtmfNumberUsed = Nothing,
+            status = exotelResponse.callStatus,
+            conversationDuration = 0,
+            recordingUrl = Nothing,
+            createdAt = now
+          }
+
+getDriverMobileNumber :: (EncFlow m r, CoreMetrics m, CacheFlow m r, EsqDBFlow m r) => (Id Person.Person, Id DM.Merchant) -> Text -> m GetDriverMobileNumberResp
+getDriverMobileNumber (driverId, merchantId) rcNo = do
+  vehicleRC <- RCQuery.findLastVehicleRC rcNo >>= fromMaybeM (InvalidRequest "Missing vehicle entry")
+  rcAssociation <- DAQuery.getActiveAssociationByRC vehicleRC.id >>= fromMaybeM (InvalidRequest "Rc not active")
+  oldDriver <- PSQuery.findById rcAssociation.driverId >>= fromMaybeM (InvalidRequest "Person not found")
+  newDriver <- PSQuery.findById driverId >>= fromMaybeM (InvalidRequest "Person not found")
+  callStatusId <- generateGUID
+  let callReq =
+        InitiateCallReq
+          { fromPhoneNum = fromMaybe "" newDriver.unencryptedMobileNumber,
```

Comment:
Don't use `unencryptedMobileNumber` number. Use `mobileNumber` only and decrypt it.